plaso/parsers/spotlight_storedb.py

```diff
@@ -0,0 +1,1050 @@
+
+      page_data = zlib.decompress(page_data)
+
+    return page_header, page_data
+
+  def _ReadVariableSizeInteger(self, data):
+    """Reads a variable size integer.
+
+    Args:
+      data (bytes): data.
+
+    Returns:
+      tuple[int, int]: integer value and number of bytes read.
+    """
+    byte_value = data[0]
+    bytes_read = 1
+
+    number_of_additional_bytes = 0
+    for bitmask in (0x80, 0xc0, 0xe0, 0xf0, 0xf8, 0xfc, 0xfe, 0xff):
+      if byte_value & bitmask != bitmask:
+        break
+      number_of_additional_bytes += 1
+
+    if number_of_additional_bytes > 4:
+      byte_value = 0
+    elif number_of_additional_bytes > 0:
+      byte_value &= bitmask ^ 0xff
+
+    integer_value = int(byte_value)
+    while number_of_additional_bytes > 0:
+      integer_value <<= 8
+
+      integer_value += int(data[bytes_read])
+      bytes_read += 1
+
+      number_of_additional_bytes -= 1
+
+    return integer_value, bytes_read
+
+  def _ReadVariableSizeIntegers(self, data, names):
+    """Reads variable size integers.
+
+    Args:
+      data (bytes): data.
+      names (list[str]): names to identify the integer values.
+
+    Returns:
+      tuple[dict[str, int], int]: integer values per name and number of bytes
+          read.
+    """
+    values = {}
+
+    data_offset = 0
+    for name in names:
+      integer_value, bytes_read = self._ReadVariableSizeInteger(
+          data[data_offset:])
+
+      data_offset += bytes_read
+
+      values[name] = integer_value
+
+    return values, data_offset
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: format specification.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(b'8tsd', offset=0)
+    return format_specification
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses an Apple Spotlight store database file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): a file-like object.
+
+    Raises:
+      UnableToParseFile: when the file cannot be parsed.
+    """
+    try:
+      file_header = self._ReadFileHeader(file_object)
+    except (ValueError, errors.ParseError):
+      raise errors.UnableToParseFile('Unable to parse file header.')
+
+    try:
+      self._map_values = []
+      self._ReadMapPages(
+          file_object, file_header.map_offset, file_header.map_size)
+
+      self._metadata_types = {}
+      self._ReadPropertyPages(
+          file_object, file_header.metadata_types_block_number,
+          self._metadata_types)
+
+      self._metadata_values = {}
+      self._ReadPropertyPages(
+          file_object, file_header.metadata_values_block_number,
+          self._metadata_values)
+
+      self._ReadPropertyPages(
```

Comment:
Maybe add a comment that we're consuming this page but don't know what it is for

Reply:
SG, done